dxr/static_unhashed/js/dxr.js

```diff
@@ -155,8 +155,13 @@ $(function() {
             searchUrl += '&case=' + isCaseSensitive;
         }
 
-        queryField.val(decodeURIComponent(fromQuery[1]));
+        queryField.val(decodeURIComponent(fromQuery[1].replace('+', ' ')));
         showBubble('info', viewResultsTxt.replace('{{ url }}', searchUrl));
+    } else {
+        var from_qQuery = /[?&]from_q=([^&]+)/.exec(location.search);
```

Comment:
Now that we render pages dynamically, we no longer need to do this through this fiddly JS regex claptrap. (It was a workaround from the days when we pre-rendered every HTML page and saved them to disk.) Now we can pass a "query=foo" and a "bubble=direct" or "bubble=single" querystring param and adjust the markup in the template.